packages/react-inbox/src/hooks/use-hover.ts

```diff
@@ -1,18 +1,24 @@
 import { useEffect, useRef, useState } from "react";
 
-function useHover() {
+function useHover(cb?: (event: Event) => boolean | void) {
   const [value, setValue] = useState(false);
   const ref: React.RefObject<HTMLDivElement> = useRef(null);
-  const handleMouseOver = () => setValue(true);
+  const handleMouseOver = (event: Event) => {
+    let shouldHover = true;
+    if (cb) {
+      shouldHover = cb(event) !== false;
+    }
+    setValue(shouldHover);
+  };
   const handleMouseOut = () => setValue(false);
   useEffect(
     () => {
       const node = ref.current;
       if (node) {
-        node.addEventListener("mouseenter", handleMouseOver);
+        node.addEventListener("mouseover", handleMouseOver);
         node.addEventListener("mouseleave", handleMouseOut);
```

Comment:
do we need to change this to mouseout? https://developer.mozilla.org/en-US/docs/Web/API/Element/mouseleave_event says mouseleave does not bubble up but mouseout does

Reply:
idk let me try and see if it still works, i might not even need to change it

Reply:
i didn't need to change this